clients/js/src/das.ts

```diff
@@ -0,0 +1,318 @@
+  | TransferDelegate
+  | UpdateDelegate
+  | PermanentFreezeDelegate
+  | Attributes
+  | PermanentTransferDelegate
+  | PermanentBurnDelegate {
+  return (({ basis_points, creators, rule_set, attribute_list, frozen }) => ({
+    ...(basis_points !== undefined ? { basisPoints: basis_points } : {}),
+    ...(creators !== undefined
+      ? {
+          creators: creators.map((creator: any) => ({
+            ...creator,
+            address: publicKey(creator.address),
+          })),
+        }
+      : {}),
+    ...(rule_set !== undefined ? { ruleSet: getRuleSet(rule_set) } : {}),
+    ...(attribute_list !== undefined ? { attributeList: attribute_list } : {}),
+    ...(frozen !== undefined ? { frozen } : {}),
+  }))(dasPluginData);
+}
+
+export function dasPluginsToCorePlugins(dasPlugins: Record<string, any>) {
+  return Object.keys(dasPlugins).reduce((acc: PluginsList, dasPluginKey) => {
+    const dasPlugin = dasPlugins[dasPluginKey];
+    const { authority, data, offset } = dasPlugin;
+    const authorityAddress = authority?.address;
+
+    acc = {
+      ...acc,
+      [convertSnakeCase(dasPluginKey)]: {
+        authority: {
+          type: authority.type,
+          ...(authorityAddress ? { address: publicKey(authorityAddress) } : {}),
+        },
+        ...dasPluginDataToCorePluginData(data),
+        offset: BigInt(offset),
+      },
+    };
+
+    return acc;
+  }, {});
+}
+
+export function dasAssetToCoreAssetOrCollection(
+  dasAsset: DasApiAsset
+): AssetV1 | CollectionV1 {
+  // TODO: Define types in Umi DAS client.
+  const {
+    interface: assetInterface,
+    id,
+    ownership: { owner },
+    content: {
+      metadata: { name },
+      json_uri: uri,
+    },
+    compression: { seq },
+    grouping,
+    authorities,
+    plugins,
+    executable,
+    lamports: lamps,
+    rent_epoch: rentEpoch,
+    mpl_core_info: mplCoreInfo,
+  } = dasAsset as DasApiAsset & {
+    plugins: Record<string, any>;
+    executable?: boolean;
+    lamports?: number;
+    rent_epoch?: number;
+    mpl_core_info?: {
+      num_minted: number;
+      current_supply: number;
+      plugins_json_version: number;
+    };
+  };
+  const { num_minted: numMinted = 0, current_supply: currentSize = 0 } =
+    mplCoreInfo ?? {};
+  const commonFields = {
+    publicKey: id,
+    uri,
+    name,
+    ...getAccountHeader(executable, lamps, rentEpoch),
+    ...dasPluginsToCorePlugins(plugins),
+    // pluginHeader: // TODO: Reconstruct
+  };
+
+  const isCollection =
+    assetInterface === (MPL_CORE_COLLECTION as DasApiAssetInterface);
+  if (isCollection) {
+    return {
+      ...commonFields,
+      key: Key.CollectionV1,
+      updateAuthority: authorities[0].address,
+      numMinted,
+      currentSize,
+    };
+  }
+
+  return {
+    ...commonFields,
+    key: Key.AssetV1,
+    owner,
+    seq: seq ? some(BigInt(seq)) : none(),
+    ...getUpdateAuthority(grouping[0], authorities[0]),
+  };
+}
+
+export async function searchAssets(
```

Comment:
Done. https://github.com/metaplex-foundation/mpl-core/pull/30/commits/d66ff2774087249f07359cf0f163d3f9aac10d04. Went with the Umi way atm, since it's the most straightforward one. 

Good question about providing this. As for other options:

I like option 2, but is it OK that we implicitly will be adding DAS stuff together with basic MPL Core functionality via `mplCore` Umi plugin?

Option 3, on the other hand, explicitly tells users that they use DAS. Though, `das.` might be a bit annoying for someone. I guess this option might the most optimal middleground

Reply:
Also, I'm thinking what should we expose exactly. Will anybody need some helpers that we use for data transformations? Or should we export only fetch/search functions?

Reply:
Made the changes according to option 3. https://github.com/metaplex-foundation/mpl-core/pull/30/commits/126c1768c7dc0eb22dc33b484c39e02112c0cc16